### PR TITLE
Pod scheduling tests improve pod deployment identification

### PR DESCRIPTION
**Pull request by omertuc**
See commit message

### CODE REVIEW

**Comment by stbenjam** [comment]
I think this makes sense to me, @xueqzhan I'll leave final lgtm to you

/approve

File: test/extended/scheduling/pods.go [hunk 1]
```diff
@@ -5,28 +5,96 @@ import (
...
+		// Pod is not owned by a replicaset, so it doesn't have a deployment
+		// associated with it
+		return ""
+	}
+
+	podOwnerReplicaSet := (*appsv1.ReplicaSet)(nil)
+	for _, namespaceReplicaSet := range namespaceReplicaSets.Items {
+		namespaceReplicaSet := namespaceReplicaSet
+		if namespaceReplicaSet.Name == podOwnerReplicaSetName {
+			podOwnerReplicaSet = &namespaceReplicaSet
```
[the remark this answers is not in the document]
Comment (omertuc) [reply test/extended/scheduling/pods.go:45]: Good catch

**Pull request merged**